staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -215,6 +222,7 @@ func (w *WebhookAuthorizer) Authorize(ctx context.Context, attr authorizer.Attri
 		// If at least one matchCondition successfully evaluates to FALSE,
 		// then the webhook is skipped.
 		if !matches {
+			metrics.RecordAuthorizationMatchConditionExclusion("Webhook", w.name)
```

Comment:
I *think* all three metrics should be recorded in `w.match()`

We should get at most one metric bump per webhook per Authorize() call, so the latency is the time it took to evaluate *all* the matchConditions on the webhook, and the failure count is the number of requests that encountered evaluation errors. Something like this:

```diff
diff --git a/staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go b/staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go
index 29ee0e84d15..00dad11e024 100644
--- a/staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go
+++ b/staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go
@@ -295,7 +295,16 @@ func (w *WebhookAuthorizer) match(ctx context.Context, r *authorizationv1.Subjec
 	if w.celMatcher == nil || w.celMatcher.CompilationResults == nil {
 		return true, nil
 	}
-	return w.celMatcher.Eval(ctx, r)
+	start := time.Now()
+	matches, sawErrors, err := w.celMatcher.Eval(ctx, r)
+	w.metrics.RecordAuthorizationMatchConditionEvaluation("Webhook", w.name, time.Since(start))
+	if sawErrors {
+		w.metrics.RecordAuthorizationMatchConditionEvaluationFailure("Webhook", w.name)
+	}
+	if err == nil && !matches {
+		w.metrics.RecordAuthorizationMatchConditionExclusion("Webhook", w.name)
+	}
+	return matches, err
 }
 
 func convertToSARExtra(extra map[string][]string) map[string]authorizationv1.ExtraValue {
```

that means Eval needs to return a little more info about whether it encountered any errors, even if it ultimately decided it could return a non-match anyway. Something like this:
```diff
diff --git a/staging/src/k8s.io/apiserver/pkg/authorization/cel/matcher.go b/staging/src/k8s.io/apiserver/pkg/authorization/cel/matcher.go
index 30ce5b69c99..efe8766d86e 100644
--- a/staging/src/k8s.io/apiserver/pkg/authorization/cel/matcher.go
+++ b/staging/src/k8s.io/apiserver/pkg/authorization/cel/matcher.go
@@ -31,7 +31,7 @@ type CELMatcher struct {
 }
 
 // eval evaluates the given SubjectAccessReview against all cel matchCondition expression
-func (c *CELMatcher) Eval(ctx context.Context, r *authorizationv1.SubjectAccessReview) (bool, error) {
+func (c *CELMatcher) Eval(ctx context.Context, r *authorizationv1.SubjectAccessReview) (matches bool, sawError bool, err error) {
 	var evalErrors []error
 	va := map[string]interface{}{
 		"request": convertObjectToUnstructured(&r.Spec),
@@ -54,13 +54,13 @@ func (c *CELMatcher) Eval(ctx context.Context, r *authorizationv1.SubjectAccessR
 		// If at least one matchCondition successfully evaluates to FALSE,
 		// return early
 		if !match {
-			return false, nil
+			return false, len(evalErrors) > 0, nil
 		}
 	}
 	// if there is any error, return
 	if len(evalErrors) > 0 {
-		return false, utilerrors.NewAggregate(evalErrors)
+		return false, true, utilerrors.NewAggregate(evalErrors)
 	}
 	// return ALL matchConditions evaluate to TRUE successfully without error
-	return true, nil
+	return true, false, nil
 }
```

Reply:
since we already have a `metrics` member field / interface in the authorizer, add the new methods for recording matchCondition metrics to that interface, and use the local `metrics` member instead of calling package methods